opal/datatype/opal_convertor.c

```diff
@@ -427,9 +636,12 @@ int32_t opal_convertor_set_position_nocheck( opal_convertor_t* convertor,
     } else {
         if( (0 == (*position)) || ((*position) < convertor->bConverted) ) {
             rc = opal_convertor_create_stack_at_begining( convertor, opal_datatype_local_sizes );
```

Comment:
The opal_convertor_create_stack_at_begining function is basically handling the special case set_position(0).